internal/common/baseclient/iigo.go

```diff
@@ -29,36 +31,47 @@ func (c *BaseClient) RuleProposal() string {
 }
 
 // GetClientPresidentPointer is called by IIGO to get the client's implementation of the President Role
+// COMPULSORY: ovverride to return a pointer to your own President object
 func (c *BaseClient) GetClientPresidentPointer() roles.President {
 	return &BasePresident{}
 }
 
 // GetClientJudgePointer is called by IIGO to get the client's implementation of the Judge Role
+// COMPULSORY: ovverride to return a pointer to your own Judge object
 func (c *BaseClient) GetClientJudgePointer() roles.Judge {
 	return &BaseJudge{}
 }
 
 // GetClientSpeakerPointer is called by IIGO to get the client's implementation of the Speaker Role
+// COMPULSORY: ovverride to return a pointer to your own Speaker object
 func (c *BaseClient) GetClientSpeakerPointer() roles.Speaker {
 	return &BaseSpeaker{}
 }
 
+// BASE
 func (c *BaseClient) TaxTaken(shared.Resources) {
```

Comment:
Shouldn't this be optional as it informs the agent tax has actually been taken from their resources

Reply:
Do we need this function?

Reply:
i've removed the comment for now

Reply:
Yes this function just tell the client that the exchange of resources has actually happened. Since the client may try and give a value that is more than what they currently have which causes an error and means this function isn't called. The function is optional however, and agents don't need to use if they do not want to